unattended.sh

```diff
@@ -0,0 +1,139 @@
+#!/bin/sh
+
+function error_and_exit {
+  echo -e "$1"
+  exit 1
+}
+
+function print_step_begin {
+  echo -e "\n$1..."
+}
+
+function print_step_end {
+  if [ -z "$1" ] ; then
+    echo -e "\tOK"
+  else
+    echo -e "\t$1"
+  fi
+}
+
+#
+# ANALYSIS
+#
+
+print_step_begin "System analysis"
+
+# Unattended install script only support Red Hat or compatible.
+if [ \! -e /etc/redhat-release ] ; then
+  error_and_exit "This unattended installation script only supports Red Hat compatible distributions. Please check https://documentation.centreon.com/docs/centreon/en/latest/installation/index.html for alternative installation methods."
+fi
+
+# systemd check.
+command -v systemctl > /dev/null 2>&1
+if [ "x$?" '=' x0 ] ; then
+  systemctl show > /dev/null 2>&1
+  if [ "x$?" '=' x0 ] ; then  
+    has_systemd=1
+  else
+    has_systemd=0
+  fi
+else
+  has_systemd=0
+fi
+
+print_step_end
+
+#
+# SELINUX
+#
+
+print_step_begin "SELinux deactivation"
+if [ -e /etc/selinux/config ] ; then
+  sed -i -e 's/^SELINUX=.*$/SELINUX=disabled/' /etc/selinux/config
+fi
+command -v selinuxenabled > /dev/null 2>&1
+if [ "x$?" '=' x0 ] ; then
+  selinuxenabled
+  if [ "x$?" '=' x0 ] ; then
+    setenforce 0
+    if [ "x$?" '!=' x0 ] ; then
+      error_and_exit "Could not disable SELinux. You might need to run this script as root."
+    fi
+    print_step_end
+  else
+    print_step_end "OK, already disabled"
+  fi
+else
+  print_step_end "OK, not detected"
+fi
+
+#
+# REPOSITORY
+#
+
+print_step_begin "Centreon official repositories installation"
+rpm -q centos-release-scl > /dev/null 2>&1
+yum -q clean all
+if [ "x$?" '!=' x0 ] ; then
```

Comment:
This condition is written to make sure that centos-scl-release is installed. If not then install.
However you are using $? which corresponds to the returned code of the previous executed command ie `yum -q clean all`

I guess that the line 77 is meant to test the returned code of the command `rpm -q centos-release-scl`

Reply:
100% truth. I fixed it by switching the lines.

Reply:
Fixed by switching lines.